compiler/rustc_middle/src/ty/query/plumbing.rs

```diff
@@ -259,52 +256,54 @@ macro_rules! define_queries {
 
         #[allow(nonstandard_style)]
         #[derive(Clone, Debug)]
-        pub enum Query<$tcx> {
-            $($(#[$attr])* $name($($K)*)),*
+        pub struct Query {
```

Comment:
I think a comment explaining what this struct is would be helpful. As far as I can tell, it's not really the query - that would imply a single instance for each (statically declared) query. Rather, this represents a particular invocation of the query (with some key), right? It's also an interesting mix of static data (like the name) and the dynamic data (hash, description, span) which we get at runtime from the key(s).

Reply:
Renamed the struct to `QueryStackFrame` and added a comment.